### PR TITLE
Lubby2 MFront

### PR DESCRIPTION
**Pull request by nagelt**
This adds a Lubby2 implementation in MFront.
Test confirms results corresponding to native implementation.

Edit: Another version with reduced size of the equation system has been added.

### CODE REVIEW

File: Tests/Data/Mechanics/Burgers/cube_1e0_mfront_mod.prj [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+                <prefix>sd</prefix>
+                <parameters>-sd_ksp_type cg -sd_pc_type bjacobi -sd_ksp_rtol 1e-16 -sd_ksp_max_it 10000</parameters>
+            </petsc>
+        </linear_solver>
+    </linear_solvers>
+    <test_definition>
+        <vtkdiff>
+            <file>cube_1e0_mfront_mod_pcs_0_ts_1_t_0.000100.vtu</file>
+            <field>displacement</field>
+            <absolute_tolerance>1e-16</absolute_tolerance>
```
Comment (endJunction) [inline Tests/Data/Mechanics/Burgers/cube_1e0_mfront_mod.prj:241]:
```suggestion
            <absolute_tolerance>5e-16</absolute_tolerance>
```

File: Tests/Data/Mechanics/Burgers/cube_1e0_mfront_mod.prj [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+        <vtkdiff>
+            <file>cube_1e0_mfront_mod_pcs_0_ts_1_t_0.000100.vtu</file>
+            <field>displacement</field>
+            <absolute_tolerance>5e-16</absolute_tolerance>
+            <relative_tolerance>1e-16</relative_tolerance>
+        </vtkdiff>
+        <vtkdiff>
+            <file>cube_1e0_mfront_mod_pcs_0_ts_101_t_1.000000.vtu</file>
+            <field>displacement</field>
+            <absolute_tolerance>1e-16</absolute_tolerance>
```
Comment (endJunction) [inline Tests/Data/Mechanics/Burgers/cube_1e0_mfront_mod.prj:247]:
```suggestion
            <absolute_tolerance>5e-16</absolute_tolerance>
```

**Comment by endJunction** [comment]
@bilke Can be merged, unless you want to inspect the pre-commit file-names error.

**Comment by zhangning737** [comment]
Good performance for Lubby2_Mod, testing cases work well...

**Comment by nagelt** [comment]
@endJunction ctests also pass with PardisoLU

**Comment by nagelt** [comment]
@zhangning737 Thanks for testing!

**Pull request merged**